### ISSUE TITLE
Is it possible to merge two DALIGenericIterator like in torch.utils.data.ConcatDataset?

### ISSUE THREAD

**Issue by dazzle-me** [NVIDIA/DALI#2841]
Suppose we have two different paths to two different folders with their jsons.
How can I "merge" them?

What I want to do is to use them as one DALIGenericIterator. 

How I would've done it if I've used torch : 
1) Create 2 ```torch.utils.data.Dataset``` objects
2) Merge them via ```torch.utils.data.ConcatDataset```
3) Create dataloaders

Any suggestions?

**Comment by JanuszL** [comment]
Hi,
It depends on the reader. If you use ['file' reader](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/supported_ops.html#nvidia.dali.fn.readers.file) the best if to manually list all files from all folders then pass it as one big list to `files` argument.
Is that what you are asking for?

**Comment by dazzle-me** [comment]
Well, I'm using COCOReader for this purpose which has ```file_root``` and ```annotations_files``` arguments.

Kind of what I want to have in the end is the ability specify **multiple** ```file_root``` and ```annotations_files``` arguments so that data can be loaded from different folders/different annotation files, because my annotations have pathes to images **relative** to the directory in which data is placed.

``` bash
big_dir/
├── data_0/
│   ├── annotations_0.json
│   └── data/
│   ...    
└── data_1/
    ├── annotations_1.json
    └── data/
    ...
```   
So, in the given example ```annotations_0.json```  has image's paths of which are relative to ```data_0```

My motivation for this move is to never modify annotations, - regardless of where the, for example, ```data_0``` is placed, annotation will always contain the right path to images which can be easily derived via 
```os.path.join(path_to_data_0, some_path_from_annotation_0.json)``` 

With current behaviour I can specify only one ```file_root = data_0/```, and hence I have to modify ```annotations_1.json``` such that every path in it has prefix ```../data_1/```.  

I'm not sure if DALI COCOReader supports such behaviour now, and if yes, I hope you caught my thought and can help to find a workaround.

Not sure that your approach works in this case because it is not clear how to assign labels then, because they're in COCO format.

**Comment by JanuszL** [comment]
Hi,

DALI doesn't support this. Adding support for multiple JSON annotation files doesn't seem to be difficult - adding a loop inside [coco_loader.cc](https://github.com/NVIDIA/DALI/blob/master/dali/operators/reader/loader/coco_loader.cc#L375) should mostly do.
The difficult part would be to support multiple file roots - the COCO reader is based on the file reader and the file reader supports only one file root. To make this possible COCO reader would need to be able to bound the file_root with the annotation. A possible solution would be to provide an empty file root to the file_label_loader and keep the full path at the coco_loader level.
If you like to dive into DALI internals we would be more than happy to review and accept any PR adding such functionality.
Otherwise, you need to merge annotation files on your own and pass it to DALI as the one entity.
